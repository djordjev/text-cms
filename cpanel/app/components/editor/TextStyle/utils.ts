import { BaseEditor, Editor, Element } from 'slate';
import { ReactEditor } from 'slate-react';

import { CustomElement } from '~/components/editor/VariationEditor/types';

const getSelectedStyle = (editor: BaseEditor & ReactEditor) => {
  const { selection } = editor;

  if (!selection) return;

  const [match] = Array.from(
    Editor.nodes<CustomElement>(editor, {
      at: selection,
      match: (n) =>
        !Editor.isEditor(n) &&
        Element.isElement(n) &&
        (n.type === 'paragraph' || n.type === 'heading')
    })
  );

  const [nodeType] = match;

  return nodeType;
};

export { getSelectedStyle };
