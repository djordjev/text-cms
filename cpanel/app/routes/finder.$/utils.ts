import { CreateParam } from '~/types';

const getCreateOpenModal = (params: URLSearchParams): CreateParam | null => {
  const param = params.get('action');

  if (param === 'new-file' || param === 'new-folder') return param;

  return null;
};

export { getCreateOpenModal };
