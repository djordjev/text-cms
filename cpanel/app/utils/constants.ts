export const COLORS = {
  BLACK: '#000',
  PRIMARY: '#003EB3'
};

export const BUTTON_ACTION = '_action';
