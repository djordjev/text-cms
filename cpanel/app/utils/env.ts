const readEnvVariable = <T = string>(name: string) => {
  const vars = process.env;

  const val = vars[name];
  if (!val) throw new Error(`Variable ${name} not supplied`);

  return val as T;
};

export { readEnvVariable };
