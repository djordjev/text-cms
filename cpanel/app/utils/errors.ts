const buildErrorResponse = (status = 404, message = 'Not found') => {
  return new Response(message, { status, statusText: message });
};

export { buildErrorResponse };
