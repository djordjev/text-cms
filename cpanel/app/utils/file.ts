const isHome = (name: string) => {
  return name.toLowerCase() === 'finder';
};

const isFile = (name: string) => {
  return name.endsWith('.txt');
};

const isFolder = (name: string) => {
  return !isFile(name);
};

const getPathFromSegments = (segments: string[], segment: string) => {
  const index = segments.indexOf(segment);

  if (index < 1) return '/finder';

  const mapped = segments.slice(0, index + 1).map((segment) => {
    if (isHome(segment)) return 'finder';

    return segment;
  });

  const joined = mapped.join('/');

  return `/${joined}`;
};

export { getPathFromSegments, isHome, isFile, isFolder };
