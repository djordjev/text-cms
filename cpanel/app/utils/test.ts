import { Descendant } from 'slate';

export const TEST_EDITOR: Descendant[] = [
  {
    type: 'heading',
    children: [
      {
        text: 'HEADING'
      }
    ],
    level: 3,
    align: 'center'
  },
  {
    type: 'paragraph',
    children: [
      {
        text: 'A '
      },
      {
        text: 'line',
        italic: true
      },
      {
        text: ' of '
      },
      {
        text: 'text',
        color: '#e41111',
        bold: true
      },
      {
        text: ' in a '
      },
      {
        text: 'paragraph',
        bold: true
      },
      {
        text: '.'
      }
    ]
  }
];

export const buildRequest = (data: Record<string, string>, method = 'POST') => {
  const form = new URLSearchParams(data);

  return new Request('https://www.test.local', { body: form, method });
};
