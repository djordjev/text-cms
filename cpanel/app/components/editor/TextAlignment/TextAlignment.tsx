import {
  IconAlignCenter,
  IconAlignLeft,
  IconAlignRight
} from '@tabler/icons-react';
import classnames from 'classnames';
import { FC } from 'react';
import { useSlate } from 'slate-react';

import { getSelectedStyle } from '~/components/editor/TextStyle/helpers';
import { Alignment, CustomElement } from '~/types/editor';

export interface TextAlignmentProps {
  className?: string;
  onChange: (selected: Partial<CustomElement>) => void;
}

const TextAlignment: FC<TextAlignmentProps> = (props) => {
  const { className, onChange } = props;

  // Hooks
  const editor = useSlate();
  const selectedStyle = getSelectedStyle(editor);
  const selected = selectedStyle?.align ?? 'left';

  // Styles
  const classesLeft = classnames('u-p-1xs', {
    'u-bg-accent u-rounded-md': selected === 'left'
  });

  const classesCenter = classnames('u-p-1xs', {
    'u-bg-accent u-rounded-md': selected === 'center'
  });

  const classesRight = classnames('u-p-1xs', {
    'u-bg-accent u-rounded-md': selected === 'right'
  });

  // Handlers
  const onChangeHandler = (alignment: Alignment) => () => {
    onChange({ align: alignment });
  };

  return (
    <div className={className}>
      <button
        aria-label="Left"
        className={classesLeft}
        onClick={onChangeHandler('left')}
        type="button"
      >
        <IconAlignLeft />
      </button>
      <button
        aria-label="Center"
        className={classesCenter}
        onClick={onChangeHandler('center')}
        type="button"
      >
        <IconAlignCenter />
      </button>
      <button
        aria-label="Right"
        className={classesRight}
        onClick={onChangeHandler('right')}
        type="button"
      >
        <IconAlignRight />
      </button>
    </div>
  );
};

export { TextAlignment };
