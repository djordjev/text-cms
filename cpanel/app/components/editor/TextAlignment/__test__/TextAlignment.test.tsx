import { createRemixStub } from '@remix-run/testing';
import { fireEvent, render, screen } from '@testing-library/react';
import { createEditor } from 'slate';
import { Slate, withReact } from 'slate-react';

import { TEST_EDITOR } from '~/utils/test';

import { TextAlignment, TextAlignmentProps } from '../TextAlignment';

test('TextAlignment', () => {
  const onChange = vi.fn();
  const editor = withReact(createEditor());

  const props: TextAlignmentProps = { onChange };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      {
        path: '/',
        Component: () => (
          <Slate editor={editor} initialValue={TEST_EDITOR}>
            <TextAlignment {...props} />{' '}
          </Slate>
        )
      }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(container.childNodes).toMatchSnapshot();

  fireEvent.click(screen.getByRole('button', { name: 'Right' }));

  expect(onChange).toHaveBeenCalledOnce();
  expect(onChange).toHaveBeenCalledWith({ align: 'right' });
});
