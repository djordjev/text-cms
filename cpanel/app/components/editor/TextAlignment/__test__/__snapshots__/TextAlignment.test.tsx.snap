// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextAlignment 1`] = `
NodeList [
  <div>
    <button
      aria-label="Left"
      class="u-p-1xs u-bg-accent u-rounded-md"
      type="button"
    >
      <svg
        class="tabler-icon tabler-icon-align-left"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6l16 0"
        />
        <path
          d="M4 12l10 0"
        />
        <path
          d="M4 18l14 0"
        />
      </svg>
    </button>
    <button
      aria-label="Center"
      class="u-p-1xs"
      type="button"
    >
      <svg
        class="tabler-icon tabler-icon-align-center"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6l16 0"
        />
        <path
          d="M8 12l8 0"
        />
        <path
          d="M6 18l12 0"
        />
      </svg>
    </button>
    <button
      aria-label="Right"
      class="u-p-1xs"
      type="button"
    >
      <svg
        class="tabler-icon tabler-icon-align-right"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6l16 0"
        />
        <path
          d="M10 12l10 0"
        />
        <path
          d="M6 18l14 0"
        />
      </svg>
    </button>
  </div>,
   ,
]
`;
